linera-chain/src/test/http_server.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! A simple HTTP server to use for testing.
+
+use std::{future::IntoFuture, net::Ipv4Addr};
+
+use axum::Router;
+use futures::FutureExt as _;
+use tokio::{net::TcpListener, sync::oneshot};
+
+/// A handle to a running HTTP server.
+///
+/// The server is gracefully shutdown when this handle is dropped.
+pub struct HttpServer {
```

Comment:
I vageuly remember seeing this already in one of your previous PRs 🤔

Reply:
Yes, in #3509. I'll resume work on that one soon, and use the type created here instead.